src/py42/_internal/clients/alertrules.py

```diff
@@ -0,0 +1,161 @@
+            user_id (str): User id who needs to be added to the criteria.
+            user_aliases (str): Email ids/Aliases who needs to be added to criteria.
+
+        Returns
+            :class:`py42.response.Py42Response`
+        """
+        tenant_id = self._user_context.get_current_tenant_id()
+        data = {
+            u"tenantId": tenant_id,
+            u"ruleId": rule_id,
+            u"userList": [
+                {
+                    # not sure what should be passed here and its significance.
+                    u"userIdFromAuthority": user_id,
+                    u"userAliasList": [user_aliases],
+                }
+            ],
+        }
+        uri = u"{0}{1}".format(self._api_prefix, u"add-users")
+        return self._session.post(uri, data=json.dumps(data))
+
+    def remove_users(self, rule_id, user_ids):
+        """Update alert rule criteria to remove users from the rule's user criteria.
+
+        Args:
+            rule_id (str): Id of a rule to be updated.
+            user_ids (str): List of comma separated user ids who needs to be added to the criteria.
+
+        Returns
+            :class:`py42.response.Py42Response`
+        """
+        tenant_id = self._user_context.get_current_tenant_id()
+        data = {u"tenantId": tenant_id, u"ruleId": rule_id, u"userIdList": [user_ids]}
+        uri = u"{0}{1}".format(self._api_prefix, u"remove-users")
+        return self._session.post(uri, data=json.dumps(data))
+        pass
+
+    def remove_all_users(self, rule_id):
+        """Update alert rule criteria to remove all users the from the rule's user criteria.
+
+        Args:
+            rule_id (str): Id of a rule to be updated.
+
+        Returns
+            :class:`py42.response.Py42Response`
+        """
+        tenant_id = self._user_context.get_current_tenant_id()
+        data = {u"tenantId": tenant_id, u"ruleId": rule_id}
+        uri = u"{0}{1}".format(self._api_prefix, u"remove-all-users")
+        return self._session.post(uri, data=json.dumps(data))
+
+    def get(self, rule_type, rule_id):
+        """Fetch alert rules by rule id.
+
+        Args:
+            rule_id (str): Id of a rule to be updated.
+            rule_type (str): Either of 'exfiltration', 'cloudshare', 'typemismatch' where
+              'exfiltration' implies rules related to movement of files,
+              'cloudshare' implies rules related to sharing/providing public access to files
+              'typemismatch' implies rules related to content and file extension mismatch.
+
+        Returns
+            :class:`py42.response.Py42Response`
+        """
+        tenant_id = self._user_context.get_current_tenant_id()
+        data = {"tenantId": tenant_id, "ruleIds": [rule_id]}
+        uri = u"{0}{1}".format(self._api_prefix, _ALERT_RULE_ENDPOINTS[rule_type])
+        return self._session.post(uri, data=json.dumps(data))
+
+
+class AlertRulesManagerClient(BaseClient):
+    """A client to manage Alert Rules."""
```

Comment:
Added both client in same file, because of lack of proper naming ideas.

Reply:
Do we need to have two different clients?

Reply:
As we need connection to 2 different hosts, we need this. I had raised a query yesterday in this context on Slack, as for different methods we need connection to different servers.